### ISSUE TITLE
Infix operator definitions incorrectly highlighted

### ISSUE THREAD

**Issue by dag** [dag/vim2hs#32]
``` haskell
f $ x = f x
```

Highlights as a function definition for `f` rather than `$`. Getting this right might be difficult, though:

``` haskell
map (x : xs) = ...
```

_should_ highlight `map` rather than `:`.

**Comment by danr** [comment]
If you decide to implement this, remember that this is also allowed:

```
(f . g) x = f (g x)
```

Extensions such as ViewPatterns and NPlusKPatterns could of course make this even trickier.

**Comment by dag** [comment]
Yeah. For the Big Rewrite I'm thinking of scrubbing this highlighting all together.  There's too many edge cases and in fact `=` is not "function definition" in Haskell so trying to highlight it as such is doomed.  For example you can do `(a,b) = ...` at the top level.
